chain/chunks/src/lib.rs

```diff
@@ -648,7 +648,13 @@ impl ShardsManager {
         for (target_account, part_ords) in bp_to_parts {
             // extra check that we are not sending request to ourselves.
             if no_account_id || me != target_account.as_ref() {
-                let parts_count = part_ords.len();
+                let prefer_peer = request_from_archival || rand::thread_rng().gen::<bool>();
+                debug!(
+                    target: "chunks", "Requesting parts {:?} for shard {} from {:?} preferring {}",
+                    &part_ords, shard_id, &target_account,
+                    if prefer_peer { "directly connected peer" } else { "routed message to target account" }
```

Comment:
Since you're already here, consider adjusting this to use KVs as [described in the style guide](https://github.com/near/nearcore/blob/master/docs/practices/style.md#tracing).